crates/matrix-sdk-ui/src/timeline/inner/state.rs

```diff
@@ -715,6 +719,12 @@ impl TimelineInnerStateTransaction<'_> {
                     self.all_events.iter_mut().find(|e| e.event_id == event_meta.event_id)
                 {
                     event.visible = event_meta.visible;
+
+                    if settings.track_read_receipts {
```

Comment:
`&& !event.visible`?

Reply:
But `Update`s only happen on previously-encrypted items, which are always visible, right?

Reply:
Ah right. Aren't you afraid that we are gonna forget this if the second case becomes possible?

Reply:
Hm, because we change how UTDs are stored such that they can be invisible and still be retried? Maybe I should be 😄

In that case, can you do

```rust
let prev_visible = mem::replace(&mut event.visible, event_meta.visible);

if settings.track_read_receipts && event.visible != prev_visible {
```

?

Reply:
We could wrap the mutation too inside the if? We don't need to update the value if it's the same.

Reply:
Done